src/nuclide_header.F90

```diff
@@ -265,11 +267,31 @@ subroutine nuclide_from_hdf5(this, group_id, temperature, method, tolerance)
                &for " // trim(this % name) // " at or near " // &
                trim(to_str(nint(temp_desired))) // " K.")
         end if
-      end do TEMP_LOOP
+      end do
 
     case (TEMPERATURE_INTERPOLATION)
-      ! TODO: Get bounding temperatures
-      call fatal_error("Temperature interpolation not yet implemented")
+      ! If temperature interpolation or multipole is selected, get a list of
+      ! bounding temperatures for each actual temperature present in the model
+      TEMP_LOOP: do i = 1, temperature % size()
+        temp_desired = temperature % data(i)
+
+        do j = 1, size(temps_available) - 1
```

Comment:
I've adopted your first easy option. The code will now revert to nearest temperature if only one temperature is available (and the user chose interpolation -- it's ok for only one temp to be present for multipole).